components/utils/src/fs.rs

```diff
@@ -86,7 +86,12 @@ pub fn copy_file_if_needed(src: &Path, dest: &Path, hard_link: bool) -> Result<(
     }
 
     if hard_link {
-        std::fs::hard_link(src, dest)?
+        if dest.exists() {
+            std::fs::remove_file(dest)
+                .with_context(|| format!("Error removing file, dst: {:?}", dest))?;
```

Comment:
dst is not going to be clear to the end error, just put `Error removing file: {:?}`